### PR TITLE
ENT-1696 Django models for authz logic on api endpoints

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
Description: This PR adds Django models for authz logic on api endpoints

JIRA: https://openedx.atlassian.net/browse/ENT-1696

Created as a task breakup of https://github.com/edx/ecommerce/pull/2176

### CODE REVIEW

File: ecommerce/core/constants.py [hunk 1]
```diff
@@ -31,6 +31,8 @@
 # Discovery Service constants
 DEFAULT_CATALOG_PAGE_SIZE = 100
 
+ENTERPRISE_COUPON_ADMIN_ROLE = 'EnterpriseCouponAdmin'
```
Comment (brittneyexline) [inline ecommerce/core/constants.py]: for consistency, let's have the constant map to `enterprise_coupon_admin`

**Review by brittneyexline — APPROVED** [review]
just one small thing i'd like to see updated before i merge!

File: ecommerce/core/constants.py (hunk 1 above)
Comment (irfanuddinahmad) [reply ecommerce/core/constants.py]: Done

**Review by christopappas — APPROVED** [review]

**Comment by irfanuddinahmad** [comment]
Closing since these changes are part of #2181

**Pull request closed without merging**